### ISSUE TITLE
[Android] BackButton no longer working

### ISSUE THREAD

**Issue by queequac** [unoplatform/uno#6617]
After upgrading from 3.8.13 to 3.91, the back button is no longer working under Android.

## Current behavior

The event does no longer get fired.

## Expected behavior

Event fires as it did all the previous versions. ;)

## How to reproduce it (as minimally and precisely as possible)

`SystemNavigationManager.GetForCurrentView().BackRequested += OnBackRequested
`

After updating to 3.9.1, the event handler is no longer called.

## Workaround

Downgrade to 3.8.13 for the moment...

## Environment

Nuget Package:
- [X] Uno.UI / Uno.UI.WebAssembly / Uno.UI.Skia
- [ ] Uno.WinUI / Uno.WinUI.WebAssembly / Uno.WinUI.Skia
- [ ] Uno.SourceGenerationTasks
- [ ] Uno.UI.RemoteControl / Uno.WinUI.RemoteControl
- [ ] Other: 

Nuget Package Version(s):
3.9.1

Affected platform(s):

- [ ] iOS
- [X] Android
- [ ] WebAssembly
- [ ] WebAssembly renderers for Xamarin.Forms
- [ ] macOS
- [ ] Skia
  - [ ] WPF
  - [ ] GTK (Linux)
  - [ ] Tizen
- [ ] Windows
- [ ] Build tasks
- [ ] Solution Templates

IDE:

- [ ] Visual Studio 2017 (version: )
- [X] Visual Studio 2019 (version: )
- [ ] Visual Studio for Mac (version: )
- [ ] Rider Windows (version: )
- [ ] Rider macOS (version: )
- [ ] Visual Studio Code (version: )

**Comment by MartinZikmund** [comment]
@queequac Can you please create some small repro so we can compare the behavior? Thanks!

**Comment by queequac** [comment]
@MartinZikmund The sample is already in the comment above. It's as easy as that. Create a new Uno multi-platform app (which results already in Uno 3.9.1 on my machine).
Add the following lines:
 ```
public MainPage()
{
    this.InitializeComponent();

    SystemNavigationManager.GetForCurrentView().BackRequested += MainPage_BackRequested;
}

private void MainPage_BackRequested(object sender, BackRequestedEventArgs e)
{
    MessageDialog dlg = new MessageDialog("Back!");
    _ = dlg.ShowAsync();
    e.Handled = true;
}
```
Start the app, press the back-button... nothing happens...
Downgrade Uno to 3.8.13, the backbutton shows the message.

The event handler is actually never called with 3.9.1. Tested with debug and release targets, in the emulator (API 28) and on physical phones (API 30), the behavior is the same.

**Comment by MartinZikmund** [comment]
For a temporary workaround you can override `OnKeyDown` in the `MainActivity` and respond to `KeyCode` of Back button by navigating back or finishing the activity. PR with fix should be merged soon.

**Comment by queequac** [comment]
@MartinZikmund Thanks for the upcoming fix! I will simply stick to 3.8.13 in the meanwhile, since a temporary workaround is not worth to be included in my app's release version on Playstore. :)
